tests/test_approve.py

```diff
@@ -817,57 +817,53 @@ def f_patch(*_args: Any, **_kwds: Any) -> NoReturn:
     assert "Unable to mark job 1 as acceptable for incident 1" in caplog.text
 
 
-def test_validate_job_qam_no_qam_data(mocker: MockerFixture) -> None:
-    approver_instance = Approver(args)
-    mocker.patch("openqabot.approver.get_json", return_value=None)
-    assert not approver_instance.validate_job_qam(1)
-
-
-def test_validate_job_qam_status_not_passed(mocker: MockerFixture) -> None:
+@pytest.mark.parametrize(
+    "json_data", [pytest.param(None, id="no_qam_data"), pytest.param({"status": "failed"}, id="status_not_passed")]
+)
+def test_validate_job_qam_no_qam_data(json_data: dict, mocker: MockerFixture) -> None:
     approver_instance = Approver(args)
-    mocker.patch("openqabot.approver.get_json", return_value={"status": "failed"})
+    mocker.patch("openqabot.approver.get_json", return_value=json_data)
     assert not approver_instance.validate_job_qam(1)
 
 
-def test_was_older_job_ok_invalid_date(caplog: pytest.LogCaptureFixture) -> None:
-    approver_instance = Approver(args)
-    job = {"build": "invalid-date", "result": "passed", "id": 123}
-    oldest_build_usable = datetime.now(UTC) - timedelta(days=1)
-    regex = re.compile(r".*")
-
-    caplog.set_level(logging.INFO)
-    assert approver_instance._was_older_job_ok(1, 1, job, oldest_build_usable, regex) is None  # noqa: SLF001
-    assert "Could not parse build date invalid-da. Won't consider this job as alternative for approval." in caplog.text
-
-
-def test_was_older_job_ok_too_old(caplog: pytest.LogCaptureFixture) -> None:
+@pytest.mark.parametrize(
+    ("job", "log_message"),
+    [
+        ({"build": "invalid-date", "result": "passed", "id": 123}, "Could not parse build date"),
+        ({"build": "20200101-1", "result": "passed", "id": 123}, "Older jobs are too old"),
+    ],
+)
+def test_was_older_job_ok_invalid_date(job: dict, log_message: str, caplog: pytest.LogCaptureFixture) -> None:
     approver_instance = Approver(args)
-    job = {"build": "20200101-1", "result": "passed", "id": 123}
     oldest_build_usable = datetime.now(UTC) - timedelta(days=1)
     regex = re.compile(r".*")
 
     caplog.set_level(logging.INFO)
     assert not approver_instance._was_older_job_ok(1, 1, job, oldest_build_usable, regex)  # noqa: SLF001
-    assert (
-        "Cannot ignore aggregate failure 1 for update 1. Reason: Older jobs are too old to be considered" in caplog.text
-    )
+    assert log_message in caplog.text
 
 
-def test_was_ok_before_no_suitable_older_jobs(caplog: pytest.LogCaptureFixture, mocker: MockerFixture) -> None:
+@pytest.mark.parametrize(
+    ("older_jobs_data", "log_message"),
+    [
+        pytest.param(
+            {"build": "20240101XY", "result": "failed", "id": 123},
+            "Older usable jobs did not succeed",
+            id="no_suitable_older_jobs",
+        ),
+        pytest.param(
+            {"build": "invalid-date"},
```

Comment:
So again an invalid date? Why do we need that twice? If we need it twice it would make sense to generalize the name of this test function accordingly.